book/src/using_rusty/build_description_file.md

```diff
@@ -42,51 +49,118 @@ To link several executables `rustyc` has the option to add libraries and automat
+- `Relocatable` generates relocatable object code (for combining with other object code)
 - `Bitcode` adds bitcode alongside machine code in executable file
-- `IR` Intermediate Representation
+- `IR` intermediate `llvm` representation
+
+The compile format is specified in the build description file as follows:  `"compile_type" : "Shared"`.
+The `compile_type` keyword is optional.
+
 
-To specify which of the above mentioned compile formats is wanted, it needs to be added to the build description file as followed: `"compile_type" : "Shared"`.
+### package_commands
+The `package_commands` keyword is optional.
+
+> TODO
+
+### Example
 
-# Example
 ```json
 {
     "files" : [
         "examples/hw.st",
         "examples/hello_world.st",
-        "examples/ExternalFunctions.st"
+        "examples/ExternalFunctions.st",
+        "examples/*.dt"
     ],
     "compile_type" : "Shared",
     "output" : "proj.so",
     "libraries" : [
         {
             "name" : "iec61131std",
             "path" : "path/to/lib",
+            "package" : "Copy",
             "include_path" : [
-                "examples/hw.st"
+                "examples/lib.st"
             ]
         },
         {
             "name" : "other_lib",
             "path" : "path/to/lib",
+            "package" : "System",
             "include_path" : [
                 "examples/hello_world.st"
             ]
         }
     ]
 }
 ```
+
+## Build Parameters
+
+The `build` subcommand exposes the following optional parameters: 
+
+### `--build-location`
+
+The build location is the location all build files will be copied to. </br>
+By default the build location is the `build` folder in the root of the project.</br>
+This can be overriden with the `--build-location` command line parameter.
+
+### `--lib-location`
+
+The lib location is where all libraries marked with `Copy` will be copied. </br>
+By default it is the same as the `build-location`.</br>
+This can be overriden with the `--lib-location` command line parameter.
+
+
+## Environment Variables
+
+Environment variables can be used inside the build description file, the variables are evaluated before an entry is evaluated.
+
+In addition to externally defined variables, the build exports variables that can be referenced in the description file:
+
+### `PROJECT_ROOT`
+
+The folder containing the `plc.json` file, i.e. the root of the project.
+
+### `ARCH`
+
+The target architecture currently being built, for a multi architecture build. The value for `ARCH` will be updated for every target.
+
+### `BUILD_LOCATION`
+
+`BUILD_LOCATION` is the folder where the build will be saved. This is the value of either the [`--build-location`](#build-location) parameter or the default build location.
+
+### `LIB_LOCATION`
+
+`LIB_LOCATION` is the folder where the lib will be saved. This is the value of either the [`--lib-location`](#lib-location) parameter or the [build location](#build-location).
```

Comment:
lib location is the folder where it looks fir libs, not where it saves libs right?

Reply:
No it's where it will save libs marked as "Copy"

Reply:
ah yes ... i mixed them up :-(